src/main/java/it/gov/pagopa/pu/debtpositions/service/DebtPositionServiceImpl.java

```diff
@@ -26,22 +30,30 @@ public class DebtPositionServiceImpl implements DebtPositionService {
   private final InstallmentPIIRepository installmentRepository;
   private final TransferRepository transferRepository;
   private final DebtPositionMapper debtPositionMapper;
+  private final OrganizationService organizationService;
 
   public DebtPositionServiceImpl(DebtPositionRepository debtPositionRepository, PaymentOptionRepository paymentOptionRepository,
                                  InstallmentPIIRepository installmentRepository, TransferRepository transferRepository,
-                                 DebtPositionMapper debtPositionMapper) {
+                                 DebtPositionMapper debtPositionMapper, OrganizationService organizationService) {
     this.debtPositionRepository = debtPositionRepository;
     this.paymentOptionRepository = paymentOptionRepository;
     this.installmentRepository = installmentRepository;
     this.transferRepository = transferRepository;
     this.debtPositionMapper = debtPositionMapper;
+    this.organizationService = organizationService;
   }
 
   @Transactional
   @Override
   public DebtPositionDTO saveDebtPosition(DebtPositionDTO debtPositionDTO) {
     Pair<DebtPosition, Map<InstallmentNoPII, Installment>> mappedDebtPosition = debtPositionMapper.mapToModel(debtPositionDTO);
 
+    if (StringUtils.isBlank(debtPositionDTO.getIupdOrg())) {
+      String accessToken = SecurityUtils.getAccessToken();
+      Optional<Organization> org = organizationService.getOrganizationById(debtPositionDTO.getOrganizationId(), accessToken);
+      org.ifPresent(organization -> mappedDebtPosition.getFirst().setIupdOrg(Utilities.generateRandomIupd(organization.getOrgFiscalCode())));
```

Comment:
all contexts which are actually calling this method doesn't already know the Organization?
If yes, ask it as parameter instead!
You should reject any DebtPosition without an Organization